vividus-plugin-db/src/main/java/org/vividus/bdd/steps/db/DatabaseSteps.java

```diff
@@ -208,6 +209,56 @@ public void compareData(String sourceSqlQuery, String sourceDbKey, String target
         verifyComparisonResult(queriesStatistic, result);
     }
 
+    /**
+     * Step waits until data received from SQL request is equal to data from examples table for some duration.
+     * Actions performed in the step:
+     * <ul>
+     *     <li>executes provided SQL query against database</li>
+     *     <li>compares data received from SQL request against examples table</li>
+     *     <li>sleeps during calculated part of specified duration</li>
+     *     <li>repeats previous actions if data was not equal and seconds timeout not expired</li>
+     * </ul>
+     * @param duration Time duration to wait
+     * @param retryTimes How many times request will be retried; duration/retryTimes=timeout between requests
+     * @param sqlQuery SQL query to execute
+     * @param dbKey Key identifying the database connection
+     * @param keys Comma-separated list of column's names to map resulting tables rows
+     * @param table Rows to compare data against
+     */
+    @When("I wait for '$duration' duration retrying $retryTimes times while data from `$sqlQuery`"
+            + " executed against `$dbKey` matching rows using `$keys` is equal to data from:$table")
+    public void waitForDataAppearance(Duration duration, int retryTimes, String sqlQuery, String dbKey,
+            Set<String> keys, ExamplesTable table)
+    {
+        long durationInMillis = duration.toMillis();
+        long expectedTime = System.currentTimeMillis() + durationInMillis;
+        long pollingTimeoutMillis = durationInMillis / retryTimes;
+
+        JdbcTemplate jdbcTemplate = getJdbcTemplate(dbKey);
+        QueriesStatistic statistics = new QueriesStatistic(jdbcTemplate, jdbcTemplate);
+        Map<Object, Map<String, Object>> sourceData = hashMap(keys, table.getRows());
+        statistics.getTarget().setRowsQuantity(sourceData.size());
+
+        List<List<EntryComparisonResult>> result;
+        do
+        {
+            List<Map<String, Object>> data = jdbcTemplate.queryForList(sqlQuery);
+            statistics.getSource().setRowsQuantity(data.size());
+            Map<Object, Map<String, Object>> targetData = hashMap(keys, data.stream()
+                    .map(m -> m.entrySet()
+                               .stream()
+                               .collect(Collectors.toMap(Map.Entry::getKey, e -> String.valueOf(e.getValue())))));
+            result = compareData(statistics, sourceData, targetData);
+            if (result.isEmpty())
+            {
+                break;
+            }
+            Sleeper.sleep(pollingTimeoutMillis, TimeUnit.MILLISECONDS);
+        }
+        while (System.currentTimeMillis() <= expectedTime);
```

Comment:
Let's introduce some common class that will handle this do-while waiter logic. We have the same login in Json waiting steps

Reply:
Please, specify which class and where you propose?